tensorflow_addons/metrics/multilabel_confusion_matrix.py

```diff
@@ -0,0 +1,105 @@
+        self.num_classes = num_classes
+        self.true_positives = self.add_weight(
+            'true_positives',
+            shape=[self.num_classes],
+            initializer='zeros',
+            dtype=self.dtype)
+        self.false_positives = self.add_weight(
+            'false_positives',
+            shape=[self.num_classes],
+            initializer='zeros',
+            dtype=self.dtype)
+        self.false_negatives = self.add_weight(
+            'false_negatives',
+            shape=[self.num_classes],
+            initializer='zeros',
+            dtype=self.dtype)
+        self.true_negatives = self.add_weight(
+            'true_negatives',
+            shape=[self.num_classes],
+            initializer='zeros',
+            dtype=self.dtype)
+
+    def update_state(self, y_true, y_pred):
+        y_true = tf.cast(y_true, tf.int32)
+        y_pred = tf.cast(y_pred, tf.int32)
+
+        # true positives
+        true_positive = tf.math.count_nonzero(y_true * y_pred, 0)
+        # predictions sum
+        pred_sum = tf.math.count_nonzero(y_pred, 0)
+        # true labels sum
+        true_sum = tf.math.count_nonzero(y_true, 0)
+        # false positives
+        false_positive = pred_sum - true_positive
+        # false negatives
+        false_negative = true_sum - true_positive
+        # true negatives
+        print('in')
+        true_negative = y_true.get_shape(
+        )[0] - true_positive - false_positive - false_negative
+
+        # true positive state update
+        self.true_positives.assign_add(tf.cast(true_positive, self.dtype))
+        # false positive state update
+        self.false_positives.assign_add(tf.cast(false_positive, self.dtype))
+        # false negative state update
+        self.false_negatives.assign_add(tf.cast(false_negative, self.dtype))
+        # true negative state update
+        self.true_negatives.assign_add(tf.cast(true_negative, self.dtype))
+
+    def result(self):
+        flat_confusion_matrix = tf.convert_to_tensor([
+            self.true_negatives, self.false_positives, self.false_negatives,
+            self.true_positives
+        ])
+        confusion_matrix = tf.reshape(
+            tf.transpose(flat_confusion_matrix), [-1, 2, 2])
+
+        return confusion_matrix
+
+    def get_config(self):
+        """Returns the serializable config of the metric."""
+
+        config = {
+            "num_classes": self.num_classes,
+        }
+        base_config = super(MultiLabelConfusionMatrix, self).get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+    def reset_states(self):
+        self.true_positives.assign(np.zeros(self.num_classes), np.int32)
+        self.false_positives.assign(np.zeros(self.num_classes), np.int32)
+        self.false_negatives.assign(np.zeros(self.num_classes), np.int32)
+        self.true_negatives.assign(np.zeros(self.num_classes), np.int32)
```

Comment:
It would be ideal to infer the dtype from `self.dtype`, but defaulting to `np.int32` or `np.float32` will work as well.